nototools/mti_cmap_data.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+#
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Extract cmap data from mti phase 3 spreadsheet."""
+
+import argparse
+from os import path
+import sys
+
+from nototools import cmap_data
+from nototools import unicode_data
+
+
+def get_script_for_name(script_name):
+  if script_name in ['LGC']:
+    return script_name
+
+  code = unicode_data.script_code(script_name)
+  if code == 'Zzzz':
+    raise ValueError('cannot identify script for "%s"' % script_name)
+  return code
+
+
+def get_script_to_cmap(csvdata):
+  # Roll our own parse, the data is simple... well, mostly.
+  # Google sheets inconsistently puts ^Z in first empty cell in a column.
+  # Also, MTI puts asterisks into the data to mark some values, which is just
+  # noise for our purposes.
+  header = None
+  data = None
+  for n, r in enumerate(csvdata.splitlines()):
+    r = r.strip();
+    if not r:
+      continue
+    rowdata = r.split(',')
+    if not header:
+      header = [get_script_for_name(name) for name in rowdata]
+      ncols = len(header)
+      data = [set() for _ in range(ncols)]
+      continue
+
+    if len(rowdata) != ncols:
+      raise ValueError('row %d had %d cols but expected %d:\n"%s"' % (
+          n, len(rowdata), ncols, r))
+    for i, v in enumerate(rowdata):
+      v = v.strip()
+      if v.endswith('*'):
```

Comment:
`strip()` can also take in characters to strip as an argument, which could be useful here.